app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -0,0 +1,58 @@
+class SinglePageSubscriptionsController < ApplicationController
```

Comment:
I think you'll need to skip CSRF before `show`, as there won't be a CSRF token in the session.